web/app/scripts/workspace/workspace-wrapper.js

```diff
@@ -386,11 +386,11 @@ angular.module('biggraph')
           const oldBox = boxes[i];
           const newBox = mapping[oldBox.id];
           for (let key in oldBox.inputs) {
-            if (!oldBox.inputs.hasOwnProperty(key)) {
-              break;
+            if (!Object.prototype.hasOwnProperty.call(oldBox.inputs, key)) {
+              continue;
```

Comment:
Was this a bug? I also don't understand what this if is doing... Isn't the above for only iterates over "own" properties? And if not, how come this never caused any issue?

Reply:
> Was this a bug? I also don't understand what this if is doing... Isn't the above for only iterates over "own" properties?

I think it was a bug to `break` on the first property from the prototype. Depending on the ordering we could have ignored real entries because they came after a prototype property.

> And if not, how come this never caused any issue?

The whole thing is a theater! These are plain JavaScript objects that don't have any prototype properties. `x = {a: 1, b: 2}; for (let k in x) { console.log(k); }` just prints `a` and `b`.